### PR TITLE
[Doc][NextJS] Document creation of new component

### PR DESCRIPTION
**Pull request by AncaIO**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation update (non-breaking change; modified files are limited to the `/docs` directory)

## Checklist:

- [ ] I have read the Contributing guide.
- [ ] My code follows the code style of this project.
- [ ] My code/comments/docs fully adhere to the Code of Conduct.
- [ ] My change is a code change and it requires an update to the documentation.
- [x] My change is a documentation change and it requires an update to the navigation.

### CODE REVIEW

File: docs/data/routes/docs/nextjs/creating-components/new-component/en.md [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+title: Walkthrough: Creating a new component
+---
+# Walkthrough: Creating a New Component
+
+This walkthrough demonstrates how to create a new component based on the chosen development workflow: 
+
+* [Using the Sitecore-first development workflow](#using-the-sitecore-first-development-workflow).
+* [Using the code-first development workflow](#using-the-code-first-development-workflow).
+
+## Using the Sitecore-first development workflow
```
Comment (nickwesselman) [inline docs/data/routes/docs/nextjs/creating-components/new-component/en.md:13]: Probably need to state the assumption here that they have used the `dotnet new` getting started template with a project name `MyProject`

File: docs/data/routes/docs/nextjs/creating-components/new-component/en.md [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+7. Publish all your item changes:
+
+   - In Content Editor, on the **Publish** ribbon, click the small black arrow next to the **Publish** icon and click **Publish site**.
+   - In the **Publish Site** window, to publish your items from the Master database to the Web database, select the **Smart publish** radio button and click **Publish**.
+
+8. You can now test the output of your new rendering:
+
+   - In your rendering host, open the site home page.
+
+   - Type the following into a PowerShell terminal to get the rendering host logs: `docker-compose logs -f rendering`.
```
Comment (nickwesselman) [inline docs/data/routes/docs/nextjs/creating-components/new-component/en.md:54]: Ahh actually this won't be the case for Next.js, you'd need to open the layout service directly. I will validate steps and add to PR.

File: docs/data/routes/docs/nextjs/creating-components/new-component/en.md [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+
+1. In the Experience Editor, open `/sitecore/content/MyProject/home`.
+2. Fill in values for all fields.
+3. Click **Save** and **Publish**.
+4. Refresh `https://myproject.localhost` and see the changes.
+
+> Be aware of the following when you use the solution:
+>
+> * A publishing of the `Platform` project updates the running `cm` service.
+> * The running rendering service uses `dotnet watch` and re-compiles automatically for any changes you make. You can also run the `rendering` project directly from Visual Studio.
```
Comment (nickwesselman) [inline docs/data/routes/docs/nextjs/creating-components/new-component/en.md:117]: Actually it uses `next dev` not `dotnet watch` to achieve this for Next.js. :) But I think this note is unnecessary actually, we can remove everything in this "Be aware" section if you are OK with it.

File: docs/data/routes/docs/nextjs/creating-components/new-component/en.md [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+   > The scaffolding system will emit a simple working component. You can modify the scaffolding conventions in `src/scripts/scaffold-component.js`.
+   >
+   > The newly created component is inactive until added to a layout on a route. See step 5.
+
+3. In the application's root directory, in `sitecore/definitions/components/MyComponent.sitecore.ts`, define the component's data. For example: 
+
+   ```typescript
+   import { CommonFieldTypes, SitecoreIcon, Manifest } from '@sitecore-jss/sitecore-jss-manifest';
+   
+   export default function (manifest: Manifest): void {
```
Comment (nickwesselman) [inline docs/data/routes/docs/nextjs/creating-components/new-component/en.md:168]: Hmm we should probably match the examples at least in template structure. I'll add to PR.

**Review by nickwesselman — CHANGES_REQUESTED** [review]
I can help with some of these changes, as noted.

File: docs/data/routes/docs/nextjs/creating-components/new-component/en.md (hunk 2 above)
Comment (AncaIO) [reply docs/data/routes/docs/nextjs/creating-components/new-component/en.md:54]: Thank you.

File: docs/data/routes/docs/nextjs/creating-components/new-component/en.md (hunk 1 above)
Comment (AncaIO) [reply docs/data/routes/docs/nextjs/creating-components/new-component/en.md:13]: I added the note right below the step links.

**Comment by nickwesselman** [comment]
@AncaIO take one more look over w/ my additions?

**Pull request merged**